tracex_project/extraction/tests.py

```diff
@@ -0,0 +1,169 @@
+        time_extraction_mock.name = "time_extraction"
+        event_type_classification_mock = MagicMock()
+        event_type_classification_mock.name = "event_type_classification"
+        location_extraction_mock = MagicMock()
+        location_extraction_mock.name = "location_extraction"
+
+        config.modules = {
+            "activity_labeling": MagicMock(return_value=activity_labeling_mock),
+            "time_extraction": MagicMock(return_value=time_extraction_mock),
+            "event_type_classification": MagicMock(return_value=event_type_classification_mock),
+            "location_extraction": MagicMock(return_value=location_extraction_mock),
+        }
+
+        orchestrator = Orchestrator(config)
+        modules = orchestrator.initialize_modules()
+
+        self.assertEqual(len(modules), 4)
+        self.assertEqual(modules[0].name, "activity_labeling")
+        self.assertEqual(modules[1].name, "time_extraction")
+        self.assertEqual(modules[2].name, "event_type_classification")
+        self.assertEqual(modules[3].name, "location_extraction")
+
+
+class ActivityLabelerTests(TestCase):
+    """Test cases for the ActivityLabeler."""
+
+    def test_execute_return_value(self):
+        """Tests if the return value of the execute method always is a dataframe and if column name is as expected."""
+        test_data = ["I fell ill yesterday.", "I went to the doctor today."]
+        activity_labeler = ActivityLabeler()
+        result = activity_labeler.execute(patient_journey_sentences=test_data)
+        self.assertIsInstance(result, pd.DataFrame)
+        self.assertIn("activity", result.columns)
+        self.assertIn("sentence_id", result.columns)
+
+
+class TimeExtractorTests(TestCase):
+    """Test cases for the TimeExtractor."""
+
+    def test_execute_return_value(self):
+        """Tests if the return value of the execute method is always a dataframe and if column names are as expected."""
+        data = {
+            'activity': ['fell ill'],
+            'sentence_id': ['1']
+        }
+        patient_journey = ["1: I fell ill on June 5 and recovered on June 7."]
+        input_dataframe = pd.DataFrame(data)
+        time_extractor = TimeExtractor()
+        result = time_extractor.execute(df=input_dataframe, patient_journey_sentences=patient_journey)
+        self.assertIsInstance(result, pd.DataFrame)
+        self.assertIn("start", result.columns)
+        self.assertIn("end", result.columns)
+        self.assertIn("duration", result.columns)
+
+
+class EventTypeClassifierTests(TestCase):
+    """Test cases for the EventTypeClassifier."""
+
+    def test_execute_return_value(self):
+        """Tests if the return value of the execute method is always a dataframe and if column name is as expected."""
+        test_data = {
+            "activity": "fell ill",
+            "start": "20220601T0000",
+            "end": "20220605T0000",
+            "duration": "04:00:00",

```

Comment:
```suggestion
            "duration": "96:00:00",
```

The duration should be 4 days, not 4 hours, based on the dates.